File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = function(nums, target) {
    let match = {};

    for (let i = 0; i < nums.length; i++) {
        let num = nums[i];
        let diff = target - num;

        if (match[diff] >= 0) {
            return [match[diff], i]
        } else {
            match[num] = i
        }
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        
        for i in range(len(nums)):
          num = nums[i]
          diff = target - num
          if diff in seen:
            return [seen[diff], i]
          else:
            seen[num] = i 